Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -489,40 +490,56 @@ protected bool ValidateInteractAndFace(EntityUid user, EntityCoordinates coordin
+            CollisionGroup collisionMask)
         {
-            if (!_mapManager.TryFindGridAt(targetPosition, out var grid))
-                return null;
-
-            if (HasComp<SharedItemComponent>(target))
+            if (!HasComp<SharedItemComponent>(target))
             {
-                // Ignore anchored entities on that tile.
-                var colliding = new HashSet<EntityUid>(grid.GetAnchoredEntities(targetPosition));
-                return e => colliding.Contains(e);
+                if (!TryComp(target, out WallMountComponent? wallMount))
+                    return null;
+
+                // wall-mount exemptions may be restricted to a specific angle range.
+                if (wallMount.Arc < 360)
+                {
+                    var angle = Angle.FromWorldVec(origin.Position - targetPosition.Position);
+                    var angleDelta = (wallMount.Direction + targetRotation - angle).Reduced().FlipPositive();
+                    var inArc = angleDelta < wallMount.Arc / 2 || Math.Tau - angleDelta < wallMount.Arc / 2;
+
+                    if (!inArc)
+                        return null;
+                }
             }
 
-            if (!TryComp(target, out WallMountComponent? wallMount))
-                return null;
+            HashSet<EntityUid> ignored;
 
-            // wall-mount exemptions may be restricted to a specific angle range.
-            if (wallMount.Arc < 360)
+            if (TryComp(target, out PhysicsComponent? physics) && physics.CanCollide)
             {
-                var angle = Angle.FromWorldVec(origin.Position - targetPosition.Position);
-                var angleDelta = (wallMount.Direction + targetRotation - angle).Reduced().FlipPositive();
-                var inArc = angleDelta < wallMount.Arc / 2 || Math.Tau - angleDelta < wallMount.Arc / 2;
-
-                if (!inArc)
-                    return null;
+                // Ignore colliding entities.
+                ignored = _sharedBroadphaseSystem.GetEntitiesIntersectingBody(target, (int) collisionMask, false, physics);
+                return e => ignored.Contains(e);
             }
 
-            // Ignore anchored entities on that tile.
-            var ignored = new HashSet<EntityUid>(grid.GetAnchoredEntities(targetPosition));
+            // Some wall mounted entities don't have collision enabled. So currently there is no way to know what
+            // entities are intersecting them. Using a aabb lookup just returns all anchored entities on any
+            // intersecting tiles. So we just cut out the middle man and return all anchored entities on the target
+            // tile. This does mean anything like directional windows that happen to lie on the same tile as the target
+            // will get ignored.
+
+            if (!_mapManager.TryFindGridAt(targetPosition, out var grid))
+                return null;
+
+            ignored = new (grid.GetAnchoredEntities(targetPosition));
```

Comment:
If an item is surrounded by 4 thindows does that mean it can still be picked up?

Reply:
Ah I forgot about entities disabling collision when going to sleep.

I've now separated the ignoring into:
- Items with collision enabled ignore entities they are colliding with them (avoids items stuck in walls)
- Wall mounted components ignore any anchored entities on the same tile (and have no collision checks)
  - Ideally wall mounted entities would only ever be on full-walls, so thindows etc shouldn't matter.
  - This doesn't check if the wall-mounted entity is anchored, because some of them (signs) aren't anchored.